src/gui/mzroll/pollyintegration.cpp

```diff
@@ -0,0 +1,263 @@
+                upload_commands.append(command);
+            }
+        }
+        line = in.readLine();
+    }
+    return upload_commands;
+}
+
+
+QStringList PollyIntegration::get_project_upload_url_commands(QByteArray result2,QStringList filenames){
+    QStringList upload_commands ;
+    QList<QByteArray> test_list = result2.split('\n');
+    int size = test_list.size();
+    QByteArray url_jsons = test_list[size-2];
+    QJsonDocument doc(QJsonDocument::fromJson(url_jsons));
+    // Get JSON object
+    QJsonObject json = doc.object();
+    QVariantMap json_map = json.toVariantMap();
+    for (int i=0; i < filenames.size(); ++i){
+        QString filename = filenames.at(i);
+        QStringList test_files_list = filename.split('/');
+        int size = test_files_list.size();
+        QString new_filename = test_files_list[size-1];
+        QString url_with_wildcard =  json_map["file_upload_urls"].toString();
+        QString url_map_json = url_with_wildcard.replace("*",new_filename) ;
+        QString command= QString("mithoo upload_project_data \"%1\" \"%2\"").arg(url_map_json).arg(filename);
+        upload_commands.append(command);
+    }
+    return upload_commands;
+}
+
+QString PollyIntegration::get_urls(QByteArray result){    
+    QList<QByteArray> test_list = result.split('\n');
+    int size = test_list.size();
+    QByteArray result2 = test_list[size-2];
+    QJsonDocument doc(QJsonDocument::fromJson(result2));
+    QJsonObject json = doc.object();
+    QVariantMap json_map = json.toVariantMap();
+    QString url =  json_map["file_upload_urls"].toString();
+    return url;
+}
+
+
+
+int PollyIntegration::check_already_logged_in(){
+    int status;
+    QString command = QString("mithoo authenticate");
+    QByteArray result2 = run_qt_process(command);
+    QList<QByteArray> test_list = result2.split('\n');
+    QByteArray status_line = test_list[0];
+    if (status_line=="already logged in"){
+        status = 1;
+    }
+    else{
+        status=0;
+    }
+    return status;
+}
+
+
+int PollyIntegration::authenticate_login(QString username,QString password){
+    QString command = QString("mithoo authenticate %1 %2").arg(username).arg(password);
+    QByteArray result = run_qt_process(command);
+    int status_inside = check_already_logged_in();
+    return status_inside;
+}
+
+
+void PollyIntegration::transferData(){
+    int status = check_already_logged_in();
+    if (status==0){
+        qDebug()<<"not logged in....";
+        login_user();
+    }
+    else {
+        qDebug("user is already logged in... sending data to polly now..");
+        get_project_name();
+    }
+}
+
+
+
+void PollyIntegration::login_user(){
+    LoginForm *w=new LoginForm(_tableDockWidget->_mainwindow,this);
+    w->setModal(true);
+    w->show();
+}
+
+void PollyIntegration::get_project_name(){
+    ProjectForm *popup=new ProjectForm(_tableDockWidget->_mainwindow,this);
```

Comment:
This object is not getting destructed, it's source of memory leak- handle it

Reply:
took care of this issue